lib/matplotlib/backends/backend_qt5.py

```diff
@@ -818,6 +838,151 @@ def _reset(self):
             self._widgets[attr].setValue(value)
 
 
+class ToolbarQt(ToolContainerBase, QtWidgets.QToolBar):
+    def __init__(self, toolmanager, parent):
+        ToolContainerBase.__init__(self, toolmanager)
+        QtWidgets.QToolBar.__init__(self, parent)
+        self._toolitems = {}
+        self._groups = {}
+        self._last = None
+
+    @property
+    def _icon_extension(self):
+        if is_pyqt5():
+            return '_large.png'
+        return '.png'
+
+    def add_toolitem(
+            self, name, group, position, image_file, description, toggle):
+
+        button = QtWidgets.QToolButton(self)
+        button.setIcon(self._icon(image_file))
```

Comment:
- These don't seem to have hi-dpi enabled
  - Whatever the lightning bolt thing is, is not implimented.  

<img width="646" alt="figure_1_and_mep22_implementation_for_qt_backend_by_fariza_ _pull_request__9934_ _matplotlib_matplotlib 2" src="https://user-images.githubusercontent.com/1562854/35349998-67a8762a-00f1-11e8-9215-aa5f0214aec5.png">
<img width="648" alt="figure_1_and_4__python_testplot_py__python3_6__and_1__jklymak_valdez____leewaves17_lwregrid2neweb_and_7__jklymak_valdez____leewaves17_archivedruns_lwregrid2_python__zsh__and_testplot_py_ ___matplotlib" src="https://user-images.githubusercontent.com/1562854/35350034-88ef65e6-00f1-11e8-9fbb-e0430d95da73.png">

Reply:
For the hi-dpi I have to check, but for the lightning bolt thing, this is normal. One of the main ideas behind MEP22 and MEP27 is to homogenize the tools and what is available in each backend.
Right now that tool is only available on QT with the navigationtoolbar

This PR is just to implement the basics of MEP22 to QT
Later another PR can add that tool for QT or preferably for all the backends

Reply:
Fo hi dpi, `NavigationToolbar2QT` has some logic associated with `is_pyqt5()` that switches to `'_large.png` and calls `setIconSize`.

Reply:
Here the logic is the same, we use a `_icon_extension` attribute in `backend_bases.ToolContainerBase`.
In the case of `qt` it gets replaced with a property
```
@property
 def _icon_extension(self):
        if is_pyqt5():
            return '_large.png'
        return '.png'
```

I have to carefully check this. But you are right, this can be done in a separate PR, because it might affect all the backends and not just this one.